src/snowflake/snowpark/dataframe_reader.py

```diff
@@ -1329,11 +1317,15 @@ def _task_fetch_from_data_source(
     result = conn.cursor().execute(query).fetchall()
     columns = [col[0] for col in schema]
     df = pd.DataFrame.from_records(result, columns=columns)
+
+    # convert timestamp and date to string to avoid but in SQL
     df = df.map(
         lambda x: x.isoformat()
         if isinstance(x, (datetime.datetime, datetime.date))
         else x
     )
+    # convert binary type to object type to avoid pandas store it as string
+    df = df.map(lambda x: x.hex() if isinstance(x, (bytearray, bytes)) else x)
```

Comment:
why calling hex here, the source is binary while we stored hexed string data in the target sf table?

Reply:
yes, this is also meant to work around a bug(feature?) in azure and gcp deployment that binary in parquet will be load as string which causes problem

Reply:
let's log a jira for this bytes parquet conversion issue